From c86739a5f01f540fc46e57ec08e2d40a382b0152 Mon Sep 17 00:00:00 2001
From: Luke Shumaker <lukeshu@lukeshu.com>
Date: Sat, 17 Jun 2017 19:03:49 -0400
Subject: [PATCH 21/38] nspawn: Change where we filter the name=systemd
 hierarchy

It was being filtered out in get_v1_hierarchies(), which seems a little
silly because it is a v1 hierarchy.  Instead, filter it in
mount_legacy_cgns_supported(); the function that actually cares about
skipping it.  This is a little less efficient, since get_v1_hierarchies()
now has to do the allocations to put it in the returned set; but that's
minor and the code is clearer now.
---
 src/nspawn/nspawn-cgroup.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/nspawn/nspawn-cgroup.c b/src/nspawn/nspawn-cgroup.c
index c91779ea85..c34ae5f360 100644
--- a/src/nspawn/nspawn-cgroup.c
+++ b/src/nspawn/nspawn-cgroup.c
@@ -246,7 +246,7 @@ static int get_v1_hierarchies(Set **ret) {
 
                 *e = 0;
 
-                if (STR_IN_SET(l, "", "name=systemd"))
+                if (streq(l, ""))
                         continue;
 
                 r = set_put_strdup(controllers, l);
@@ -360,6 +360,9 @@ static int mount_legacy_cgns_supported(
                 if (!hierarchy)
                         break;
 
+                if (streq(hierarchy, "name=systemd"))
+                        continue;
+
                 r = mount_legacy_cgroup_hierarchy("", hierarchy, hierarchy, !userns);
                 if (r < 0)
                         return r;
-- 
2.18.0

